### PR TITLE
doc: Add link to generated API docs on API page

### PR DESCRIPTION
**Pull request by johanbrandhorst**
This makes it easier to find the generated API docs from the API description page.

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
@thiskevinwang is there some way I can check a test deployment of this? I'm not sure the link is correct.

**Comment by thiskevinwang** [comment]
@johanbrandhorst — Working on a change to make this possible!

(Also the link looks good to me)

File: website/content/docs/api-clients/api.mdx [hunk 1]
```diff
@@ -12,6 +12,7 @@ Boundary's API is a JSON-based HTTP API that adheres to a set of standards that
 Before reading this page, it is useful to understand Boundary's [domain model](/docs/concepts/domain-model) to be aware of the terminology used here.
 
 Boundary's API is also described via OpenAPI v2; the version corresponding to any tag of Boundary's source code can be found in Boundary's [GitHub repository](https://github.com/hashicorp/boundary/blob/main/internal/gen/controller.swagger.json).
+A rendered version of this generated API definition can be found on the [API page](/api-docs).
```
Comment (yhyakuna) [inline website/content/docs/api-clients/api.mdx:15]: If you want the link to visually stand out, you can make it into a callout note (use `->`)?  

image

Just a thought.  Feel free to ignore this comment if you don't like the idea 😅 

```suggestion

-> **NOTE:** A rendered version of this generated API definition can be found on the [API page](/api-docs).
```

**Review by thiskevinwang — APPROVED** [review]
@johanbrandhorst If you want to preview this _locally_, you can rebase/merge in `main`, and run

```bash
# from ./website
make website/build-local && PREVIEW_MODE=developer make website/local
```

Or if you want a deployed preview you can commit a change to update this value to `developer`
- https://github.com/hashicorp/boundary/blob/main/website/scripts/website-build.sh#L10

Either way the link looks good to me!

**Pull request merged**